Objective-C/CBLReplicator.mm

```diff
@@ -358,6 +358,67 @@ - (void) removeChangeListenerWithToken: (id<CBLListenerToken>)token {
     }
 }
 
+- (NSSet<NSString*>*) pendingDocumentIds: (NSError**)error {
+    if (_config.replicatorType > 1) {
+        if (error)
+            *error = [NSError errorWithDomain: CBLErrorDomain
+                                         code: CBLErrorUnsupported
+                                     userInfo: @{NSLocalizedDescriptionKey: kCBLErrorMessagePullOnlyPendingDocIDs}];
+        return nil;
+    }
+    
+    C4Error c4err = {};
+    if (![self _setupC4Replicator: &c4err]) {
```

Comment:
This starts a replicator, now?

Reply:
it will initialize a replicator, if it already not done.

Reply:
do you have to close it or anything?

Reply:
_setupC4Replicator: should be called under the lock (same lock as the start method) for thread safety.

Reply:
sure, will make the changes.